src/DomainLib/Aggregates/EventNameMapping.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace DomainLib.Aggregates
+{
+    // ReSharper disable once InconsistentNaming
+    public class EventNameMapping : IEventNameMapping
+    {
+        private readonly Dictionary<string, Type> _eventNameToTypeMappings;
+        private readonly Dictionary<Type, string> _eventTypeToNameMappings;
+
+        public EventNameMapping()
+        {
+            _eventNameToTypeMappings = new Dictionary<string, Type>();
+            _eventTypeToNameMappings = new Dictionary<Type, string>();
+        }
+        
+        public void RegisterEvent<TEvent>(bool throwOnConflict = true)
+        {
+            RegisterEvent(typeof(TEvent), throwOnConflict);
+        }
+
+        public void RegisterEvent(Type clrType, bool throwOnConflict = true)
+        {
+            var eventName = DetermineEventNameForClrType(clrType);
+            RegisterEventName(eventName, clrType, throwOnConflict);
+        }
+
+        public Type GetClrTypeForEventName(string eventName)
+        {
+            if (_eventNameToTypeMappings.TryGetValue(eventName, out var clrType))
+            {
+                return clrType;
+            }
+
+            throw new UnknownEventNameException($"{eventName} could not be mapped to a CLR type", eventName);
+        }
+
+        public string GetEventNameForClrType(Type clrType)
+        {
+            return _eventTypeToNameMappings.TryGetValue(clrType, out var eventName)
+                ? eventName
+                : DetermineEventNameForClrType(clrType);
+        }
+
+        IEnumerable<KeyValuePair<string, Type>> IEventNameMapping.GetNameToTypeMappings()
+        {
+            return _eventNameToTypeMappings.ToImmutableDictionary();
+        }
+
+        public void Merge(IEventNameMapping other, bool throwOnConflict = true)
+        {
+            foreach (var (key, value) in other.GetNameToTypeMappings())
+            {
+                RegisterEventName(key, value, throwOnConflict);
+            }
+        }
+
+        private static string DetermineEventNameForClrType(Type clrType)
+        {
```

Comment:
Should this comment be part of a proper documentation comment somewhere?

Reply:
I'll set up a proper wiki site where we can document usage properly.